scripts/tests/test_zkai_native_seq32_attention_mlp_single_proof_gate.py

```
@@ -0,0 +1,135 @@
+        self.assertEqual(summary["matched_two_proof_frontier_typed_bytes"], self.gate.FRONTIER_TYPED_BYTES)
+        self.assertEqual(summary["typed_saving_vs_matched_frontier_bytes"], self.gate.TYPED_SAVING_BYTES)
+        self.assertEqual(summary["native_single_proof_json_bytes"], self.gate.SINGLE_PROOF_JSON_BYTES)
+        self.assertEqual(summary["json_saving_vs_matched_frontier_bytes"], self.gate.PROOF_JSON_SAVING_BYTES)
+        self.assertEqual(summary["proof_size_comparable_external_rows"], 0)
+        self.gate.validate_payload(payload)
+
+    def test_rejects_erased_typed_saving(self) -> None:
+        payload = copy.deepcopy(self.__class__.payload)
+        payload["summary"]["typed_saving_vs_matched_frontier_bytes"] = 0
+        payload["payload_commitment"] = self.gate.payload_commitment(payload)
+        with self.assertRaisesRegex(self.gate.NativeSeq32AttentionMlpSingleProofGateError, "summary drift"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)